internal/rm/rm.go

```diff
@@ -110,6 +112,35 @@ func (r *resourceManager) Devices() Devices {
 	return r.devices
 }
 
+var errInvalidRequest = errors.New("invalid request")
+
+// ValidateRequests checks the requested IDs against the resource manager configuration.
+// This asserts that all requested IDs are known to the resource manager and that the request is
+// valid for a specified sharing configuration.
+// specified sharing configuration is respected.
+func (r *resourceManager) ValidateRequest(ids AnnotatedIDs) error {
+	for _, id := range ids {
+		if !r.devices.Contains(id) {
+			return fmt.Errorf("%w: unknown device: %s", errInvalidRequest, id)
+		}
+	}
+	// If the devices being allocated are replicas, then (conditionally)
+	// error out if more than one resource is being allocated.
+	includesReplicas := ids.AnyHasAnnotations()
+	numRequestedDevices := len(ids)
+	switch r.config.Sharing.SharingStrategy() {
+	case spec.SharingStrategyTimeSlicing:
+		if includesReplicas && numRequestedDevices > 1 && r.config.Sharing.ReplicatedResources().FailRequestsGreaterThanOne {
+			return fmt.Errorf("%w: maximum request size for shared resources is 1; found %d", errInvalidRequest, numRequestedDevices)
```

Comment:
I don't see the point of using a wrapped error here, as it is just a string constant. We could consider using a custom struct  that implements the `Error` interface.

Reply:
The main reason for wrapping the error here is to allow testing for the singpost errors in the unit tests. I suppose I could update the tests to check the error contents.

If you have suggestions on how to make this robust, I'd be happy to update this (possibly as a follow up).

Reply:
We could create a struct that implements the Error interface. The struct could have the relevant fields populated during the time of instantiating the error. The implementing `Error` method could print out the error message based on the struct fields